test/builder/pod.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2019 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builder
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PodOp is an operation which modifies a Pod struct.
+type PodOp func(*corev1.Pod)
+
+// PodSpecOp is an operation which modifies a PodSpec struct.
+type PodSpecOp func(*corev1.PodSpec)
+
+// Pod creates a Pod with default values.
+// Any number of Pod modifiers can be passed to transform it.
+func Pod(name, namespace string, ops ...PodOp) *corev1.Pod {
```

Comment:
I've added a builder for `Pods` since it was getting pretty hard to read what was going on in `taskrun_test.go`. Happy to back this out or split it into a separate PR if desired.

Reply:
@pivotal-nader-ziada No problem, I will take a look at splitting it out into another PR tomorrow!

Reply:
thanks!